spec/models/student_spec.rb

```diff
@@ -319,6 +327,16 @@
         create(:grace_period_deduction, membership: membership, deduction: 20)
         expect(@student.remaining_grace_credits).to eq(-25)
       end
+
+      it 'returns the correct value of used credits per assessment' do
+        create(:grace_period_deduction, membership: @membership, deduction: 2)
+        expect(@student.grace_credits_used_for(@assignment)).to eq(2)
+      end
+
+      it 'updates remaining grace credits after deductions' do
+        create(:grace_period_deduction, membership: @membership, deduction: 2)
```

Comment:
I don't like how I run `create(:grace_period_deduction, membership: @membership, deduction: 2)` two times. Putting this line in a before-do block causes the other tests in the context block to be affected, however.

Would it be more optimal to create a new context block for these two test cases?

Reply:
Yes, a context block here is good. Something like `context 'where there is a deduction'`